### DISCUSSION TITLE
Future of WindowsAppSDK runtime (self-containment)

### DISCUSSION THREAD

**Discussion by AzAgarampur** [microsoft/WindowsAppSDK#2319] [Q&A]
So I see that 1.1 has plans to include "self contained" support. This is good, as I've always wanted to be able to have unpackaged apps that "truly" don't need any sort of installation.

However, there are also future projections that WinUI 3, for example, will become a part of the Windows OS itself. Does this mean, that since WinUI 3 will be a part of the OS, the WindowAppSDK packages will start to be installed by default in Windows 11 in the future? If so, deploying apps that use WindowsAppSDK would be a lot like running .NET framework 4.X apps. These .NET framework apps didn't need any runtime install, because windows comes with the framework already installed. I'm wondering if something similar might happen in the future for WindowsAppSDK.

**Comment by dongle-the-gadget** [comment]
WinUI 3's goal is to literally decouple the UI framework from the OS.

**Reply by AzAgarampur** [reply]
It will always be "decoupled" as in not tied to platform restrictions (UWP/WinRT only, AppContainer only, etc) but might it just be "pre-installed" in the os?

*However now that I think of it this might be a problem for older WinUI 3 apps that might depend on something like version 1.1 if, for example, 3.5 is installed on the system...*

**Reply by dongle-the-gadget** [reply]
Actually .NET Core was never installed anyway, so I would doubt that would happen.
